tests/integration/replication.tcl

```diff
@@ -444,12 +442,157 @@ test {slave fails full sync and diskless load swapdb recovers it} {
                 fail "Replica didn't disconnect"
             }
 
-            # make sure the original keys were restored
+            # make sure the original keys haven't changed
             assert_equal [$slave dbsize] 2000
         }
     }
 } {} {external:skip}
 
+test {Diskless load swapdb succeeds when master replid is not same (no async_loading)} {
```

Comment:
where there any changes in the new tests since my last review (before you rebased)?
i rather review the diff and not re-read the whole thing, but i can't find the old commit in GH.

Reply:
No changes in the tests. The big rewrite is just a rebase + integrating with the 2 conflicting PRs